guides/source/active_model_basics.md

```diff
@@ -44,76 +67,331 @@ class EmailContact
+same time.
 
 ```ruby
 class Person
   include ActiveModel::AttributeMethods
 
-  attribute_method_prefix 'reset_'
-  attribute_method_suffix '_highest?'
-  define_attribute_methods 'age'
+  attribute_method_affix prefix: "reset_", suffix: "_to_default!"
+  attribute_method_prefix "first_"
+  attribute_method_prefix "last_"
+  attribute_method_suffix "_short?"
+
+  define_attribute_methods "name"
 
-  attr_accessor :age
+  attr_accessor :name
 
   private
-    def reset_attribute(attribute)
-      send("#{attribute}=", 0)
-    end
 
-    def attribute_highest?(attribute)
-      send(attribute) > 100
-    end
+  # this will create a method 'first_name'
+  def first_attribute(attribute)
+    public_send(attribute).split.first
+  end
+
+  # this will create a method 'last_name'
+  def last_attribute(attribute)
+    public_send(attribute).split.last
+  end
+
+  # this will create a method 'name_short?'
+  def attribute_short?(attribute)
+    public_send(attribute).length < 5
+  end
+
+  # this will create a method 'reset_name_to_default!'
+  def reset_attribute_to_default!(attribute)
+    public_send("#{attribute}=", "Default Name")
+  end
 end
 ```
 
 ```irb
 irb> person = Person.new
-irb> person.age = 110
-irb> person.age_highest?
-=> true
-irb> person.reset_age
-=> 0
-irb> person.age_highest?
+
+irb> person.name = "Jane Doe"
+
+# utilizing the `first_name` and `last_name` prefix methods
+irb> person.first_name
+=> "Jane"
+irb> person.last_name
+=> "Doe"
+
+# utilizing the `name_short?` suffix method
+irb> person.name_short?
 => false
+
+# utilizing the `reset_name_to_default!` prefix and suffix methods
+irb> person.reset_name_to_default!
+=> "Default Name"
+```
+
+If you call a method that is not defined, it will raise a `NoMethodError`
+error.
+
+#### Method: `alias_attribute`
+
+`ActiveModel::AttributeMethods` also provides aliasing of attribute methods.
+This can be done by using the `alias_attribute` method.
+
+```ruby
+class Person
+  include ActiveModel::AttributeMethods
+
+  attribute_method_suffix '_short?'
+  define_attribute_methods :name
+
+  attr_accessor :name
+
+  alias_attribute :nickname, :name
+
+  private
+
+  def attribute_short?(attribute)
+    public_send(attribute).length < 5
+  end
+
+end
+```
+
+```irb
+irb> person = Person.new
+
+irb> person.name = "Joe"
+
+irb> person.name
+=> "Joe"
+
+# `nickname` is the alias for `name`, and returns the same value
+irb> person.nickname
+=> "Joe"
+irb> person.name_short?
+=> true
+
+# `nickname_short?` is the alias for `name_short?`, and returns the same value
+irb> person.nickname_short?
+=> true
 ```
 
 ### Callbacks
 
-`ActiveModel::Callbacks` gives Active Record style callbacks. This provides an
-ability to define callbacks which run at appropriate times.
-After defining callbacks, you can wrap them with before, after, and around
-custom methods.
+`ActiveModel::Callbacks` gives plain Ruby objects Active Record style callbacks.
```

Comment:
Should we link to the Active Record Callbacks guide? https://guides.rubyonrails.org/active_record_callbacks.html

Can be a link, or can be an info box.